tests/integration/utils2.go

```diff
@@ -131,59 +148,57 @@ var previousTestCaseTestName string
 	//  that don't have the flag defined
-	badgerFileValue, _ := os.LookupEnv(fileBadgerEnvName)
-	badgerInMemoryValue, _ := os.LookupEnv(memoryBadgerEnvName)
-	databaseDir, _ = os.LookupEnv(fileBadgerPathEnvName)
-	rootDatabaseDir, _ = os.LookupEnv(rootDBFilePathEnvName)
-	detectDbChangesValue, _ := os.LookupEnv(detectDbChangesEnvName)
-	inMemoryStoreValue, _ := os.LookupEnv(inMemoryEnvName)
-	repositoryValue, repositorySpecified := os.LookupEnv(repositoryEnvName)
-	setupOnlyValue, _ := os.LookupEnv(setupOnlyEnvName)
-	targetBranchValue, targetBranchSpecified := os.LookupEnv(targetBranchEnvName)
-	mutType, mutationTypeSpecified := os.LookupEnv(mutationTypeEnvName)
-
-	badgerFile = getBool(badgerFileValue)
-	badgerInMemory = getBool(badgerInMemoryValue)
-	inMemoryStore = getBool(inMemoryStoreValue)
-	DetectDbChanges = getBool(detectDbChangesValue)
-	SetupOnly = getBool(setupOnlyValue)
-
-	if !repositorySpecified {
+	httpClient, _ = strconv.ParseBool(os.Getenv(clientHttpEnvName))
+	goClient, _ = strconv.ParseBool(os.Getenv(clientGoEnvName))
+	badgerFile, _ = strconv.ParseBool(os.Getenv(fileBadgerEnvName))
+	badgerInMemory, _ = strconv.ParseBool(os.Getenv(memoryBadgerEnvName))
+	inMemoryStore, _ = strconv.ParseBool(os.Getenv(inMemoryEnvName))
+	DetectDbChanges, _ = strconv.ParseBool(os.Getenv(detectDbChangesEnvName))
+	SetupOnly, _ = strconv.ParseBool(os.Getenv(setupOnlyEnvName))
+
+	var repositoryValue string
+	if value, ok := os.LookupEnv(repositoryEnvName); ok {
+		repositoryValue = value
+	} else {
 		repositoryValue = "https://github.com/sourcenetwork/defradb.git"
 	}
 
-	if !targetBranchSpecified {
+	var targetBranchValue string
+	if value, ok := os.LookupEnv(targetBranchEnvName); ok {
+		targetBranchValue = value
+	} else {
 		targetBranchValue = "develop"
 	}
 
-	if mutationTypeSpecified {
-		mutationType = MutationType(mutType)
+	if value, ok := os.LookupEnv(mutationTypeEnvName); ok {
+		mutationType = MutationType(value)
 	} else {
 		// Default to testing mutations via Collection.Save - it should be simpler and
 		// faster. We assume this is desirable when not explicitly testing any particular
 		// mutation type.
 		mutationType = CollectionSaveMutationType
 	}
 
-	// default is to run against all
-	if !badgerInMemory && !badgerFile && !inMemoryStore && !DetectDbChanges {
-		badgerInMemory = true
-		// Testing against the file system is off by default
-		badgerFile = false
-		inMemoryStore = true
-	}
-
-	if DetectDbChanges {
+	// Set default values for the specified testing mode.
+	switch {
+	case DetectDbChanges:
+		// Change detector runs using only the go client type.
+		goClient = true
+		httpClient = false
 		detectDbChangesInit(repositoryValue, targetBranchValue)
-	}
-}
 
-func getBool(val string) bool {
-	switch strings.ToLower(val) {
-	case "true":
-		return true
 	default:
-		return false
+		// Default is to test all client types.
+		if !goClient && !httpClient {
```

Comment:
todo: I disagree with this default.  It will essentially double the stuff being done when running `go test` or `make test`, and will significantly degrade the dev-ex of anyone working in the codebase.

Can you please default to the go client only, and then maybe add a couple of make actions to make it easy to run http?